src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metricsdb/MetricsDB.java

```diff
@@ -84,6 +85,15 @@ public MetricsDB(long windowStartTime) throws Exception {
     create = DSL.using(conn, SQLDialect.SQLITE);
   }
 
+  public static MetricsDB fetchExisting(long windowStartTime) throws Exception {
+    String filePath = PluginSettings.instance()
+            .getSettingValue(DB_FILE_PREFIX_PATH_CONF_NAME, DB_FILE_PREFIX_PATH_DEFAULT) + windowStartTime;
+    if (!(new File(filePath)).exists()) {
+      throw new FileNotFoundException(String.format("MetricsDB file %s could not be found.", filePath));
```

Comment:
The new changes should not crash the system. we should handle this gracefully.

Also, can we add javadoc comments to public methods ?

Reply:
This change would not crash the system. The QueryBatchRequestHandler (the only place where this is used) catches it exception and handles it appropriately. Also, this code here isn't introducing any new error cases -- it has the same abstraction of throwing an error if there is any issue accessing the file or creating a jdbc connection to it as the MetricsDB constructor.

Reply:
Agreed about the javadocs.